app/main.py

```diff
@@ -1,5 +1,71 @@
+from __future__ import annotations
+
+
+class BaseRobot:
+    def __init__(
+            self,
+            name: str,
+            weight: float,
+            coords: list[float] = None
+    ) -> None:
+        self.name = name
+        self.weight = weight
+        self.coords = coords if coords else [0, 0]
+
+    def go_forward(self, step: float = 1) -> None:
+        self.coords[1] += step
+
+    def go_back(self, step: float = 1) -> None:
+        self.coords[1] -= step
+
+    def go_right(self, step: float = 1) -> None:
+        self.coords[0] += step
+
+    def go_left(self, step: float = 1) -> None:
+        self.coords[0] -= step
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(
+            self,
+            name: str,
+            weight: float,
+            coords: list[float] = None
+    ) -> None:
+        coords = coords if coords else [0, 0, 0]
+        super().__init__(name, weight, coords)
```

Comment:
As an option, you could consider using the following syntax:
```suggestion
        super().__init__(name, weight, coords or [0, 0, 0])
```